trinity/tx_pool/validators.py

```diff
@@ -0,0 +1,59 @@
+from typing import (
+    Type
+)
+
+from evm.chains.base import (
+    BaseChain
+)
+from evm.exceptions import (
+    ValidationError
+)
+from evm.rlp.transactions import (
+    BaseTransaction,
+    BaseTransactionFields
+)
+from evm.vm.base import (
+    BaseVM
+)
+
+
+class DefaultTransactionValidator():
+    """
+    The :class:`~trinity.tx_pool.validators.DefaultTransactionValidator` class is responsible to
+    decide wether transactions should be relayed to peers or not. This implementation validates
+    transactions against an initially provided transaction class but will switch to a different
+    one as soon as the tip of the chain uses a more up to date transaction class than the one
+    that was initially provided.
+    """
+
+    def __init__(self, chain: BaseChain, initial_tx_class: Type[BaseTransaction]) -> None:
+        self.chain = chain
+        self._initial_tx_class = initial_tx_class
+        self._ordered_tx_classes = [
+            vm_class.get_transaction_class()
+            for _, vm_class in chain.get_vm_configuration()
+        ]
+        self._initial_tx_index = self._ordered_tx_classes.index(initial_tx_class)
```

Comment:
This will throw an error if the provided `initial_tx_class` is not in the `vm_configuration`.  Probably worth checking existence as I think it should be a validation error if the provided *initial* class is not found in the `vm_configuration`.

Reply:
Let's go with the block number as you suggested.